build/helper/metadata_add_all.py

```diff
@@ -54,7 +54,8 @@ def _add_python_method_name(function, name):
             function['python_name'] = '_' + camelcase_to_snakecase(name)
         else:
             function['python_name'] = camelcase_to_snakecase(name)
-            assert 'method_name_for_documentation' not in function, "'method_name_for_documentation' not allowed to be set: function['method_name_for_documentation'] = '{0}', function['python_name'] = '{1}'".format(function['method_name_for_documentation'], function['python_name'])
+            if function['codegen_method'] != 'no':
```

Comment:
turn this into a single statement...

    assert function['codegen_method'] != `no` and ....

Reply:
I've updated this line to `assert function['codegen_method'] == 'no' or 'method_name_for_documentation' not in function ...` in the latest commit. I've also updated #1709 accordingly.